File: python/02/02.py
import sys


commands = sys.stdin.read().split('\n')
commands = [[command.split()[0], int(command.split()[1])] for command in commands]

horpos, depth, aim = 0, 0, 0

for command in commands:
	direction = command[0]
	length = command[1]

	if direction == 'forward':
		horpos = horpos + length
		depth = depth + aim*length
	elif direction == 'up':
		aim = aim - length
	elif direction == 'down':
		aim = aim + length

print(depth*horpos)

File: go/02/02.go
package main

import (
	"fmt"
)

func main() {
	var command string 
	var length int
	var depth int
	horpos := 0
	aim := 0
	for {
		fmt.Scanf("%s %d", &command, &length)
		if command == "0" {
			break
		}

		switch command {
		case "forward":
			horpos += length
			depth += aim*length
		case "up":
			aim -= length
		case "down":
			aim += length
		}
	}
	
	fmt.Println(horpos*depth)
}
